### PR TITLE
Render emails on admin UI 

### PR DESCRIPTION
**Pull request by josejulio**
This could help devs into trying an email template without actually having to put the code in the repository and trigger an event.
The idea is to write the templates (subject, body) and the payloads to render on the page.
There is sample data to get started.

Screenshot_2021-08-16_11-14-49

Screenshot_2021-08-16_11-16-25

### CODE REVIEW

**Comment by gwenneg** [comment]
I can finally use my new React powers I acquired during my holidays!

test

(I worked on a personal project with a React-based UI)

**Comment by josejulio** [comment]
> I can finally use my new React powers I acquired during my holidays!

yay!

File: backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package com.redhat.cloud.notifications.routers.models;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+public class RenderEmailTemplateRequest {
```
Comment (gwenneg) [inline backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java:6]: This probably needs `@JsonNaming(SnakeCaseStrategy.class)` and `@Schema` annotations on the fields for the OpenAPI output.

File: backend/src/main/webapp/src/pages/RenderEmailPage.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+import { useRenderEmailRequest } from '../services/RenderEmailRequest';
+
+const defaultSubjectTemplate = `
+Important email to {user.firstName} from MyCoolApp!
+`.trimLeft();
+
+const defaultBodyTemplate = `
+<div>Hello {user.firstName} {user.lastName},</div>
+<div>We have some important news to you, MyApp has a notification for you</div>
```
Comment (gwenneg) [inline backend/src/main/webapp/src/pages/RenderEmailPage.tsx:24]:
```suggestion
<div>We have some important news for you, MyApp has a notification for you</div>
```

File: backend/src/main/webapp/src/pages/RenderEmailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            metadata: {},
+            payload: '{"my_id":"3df53241-3e09-481b-a322-4892caaaaadc","my_name":"Good name"}'
+        },
+        {
+            metadata: {},
+            payload: '{"my_id":"6c5e8451-a40a-4bb7-ab9a-0cb10a4c577d","my_name":"Bad name"}'
+        },
+        {
+            metadata: {},
+            payload: '{"my_id":"b4c6378a-c1fb-4d3e-8e9b-7e5bdfc09dd3","my_name":"Ugly name"}'
```
Comment (gwenneg) [inline backend/src/main/webapp/src/pages/RenderEmailPage.tsx:63]: Can we replace `Ugly name` (and maybe `Bad name`) with something more positive? 😄

**Comment by gwenneg** [comment]
This is a great feature for our customers experience with the notifications onboarding 👍

File: backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java (hunk 1 above)
Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java:6]: Yes, we need the SnakeCaseStrategy. Not sure why we need the `@Schema` 
 https://github.com/RedHatInsights/notifications-backend/pull/478/files#diff-c2092ee52cefc16270b72e1ae4f060f48567699e6cdb8eb2317d1e66c0f2d22fR348-R352

File: backend/src/main/webapp/src/pages/RenderEmailPage.tsx (hunk 3 above)
Comment (josejulio) [reply backend/src/main/webapp/src/pages/RenderEmailPage.tsx:63]: It was a joke :grin:

File: backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java (hunk 1 above)
Comment (gwenneg) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java:6]: Isn't the new API part of an OpenAPI JSON file?

Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/routers/models/RenderEmailTemplateRequest.java:10]: It is part of the internal openapi. The types are build from the java types in this case, so we don't have to put `@Schema` in here

File: backend/src/main/webapp/src/pages/RenderEmailPage.tsx (hunk 3 above)
Comment (gwenneg) [reply backend/src/main/webapp/src/pages/RenderEmailPage.tsx]: I know but I was afraid someone could take this seriously 😄

**Review by gwenneg — APPROVED** [review]
LGTM after the Checkstyle issue fix

**Pull request merged**